src/components/card/card.tsx

```diff
@@ -0,0 +1,59 @@
+type CardProps = {
+  environment: string;
+  name: string;
+  type: string;
+  link: string;
+  price: number;
+  rating: number;
+  isPremium: boolean;
+  isFavorite: boolean;
+  img: string;
+}
+
+function Card(props: CardProps) {
+  const img = {
+    width: props.environment === 'cities' ? 260 : 150,
+    height: props.environment === 'cities' ? 200 : 110
+  };
+
+  return (
+    <article className={`${props.environment}__card place-card`}>
+      {props.isPremium && (
+        <div className="place-card__mark">
+          <span>Premium</span>
+        </div>)}
+
+      <div className={`${props.environment}__image-wrapper place-card__image-wrapper`}>
+        <a href={props.link}>
+          <img className="place-card__image" src={props.img} width={img.width} height={img.height} alt={props.name} />
+        </a>
+      </div>
+      <div className={`${props.environment}__card-info place-card__info`}>
+        <div className="place-card__price-wrapper">
+          <div className="place-card__price">
+            <b className="place-card__price-value">&euro;{props.price}</b>
+            <span className="place-card__price-text">&#47;&nbsp;night</span>
+          </div>
+          <button className={`place-card__bookmark-button button ${props.isFavorite ? 'place-card__bookmark-button--active' : ''}`} type="button">
```

Comment:
Булевые CSS классы обычно через [classnames ](https://www.npmjs.com/package/classnames )пишутся. А ещё лучше
```suggestion
    <button className={classNames('place-card__bookmark-button', 'button', {
            'place-card__bookmark-button--active': props.isFavorite
          })} type="button"
          >
```

А ещё лучше [clsx](https://www.npmjs.com/package/clsx)